x11-libs/wxgtk/wxgtk-3.1.5.recipe

```diff
@@ -0,0 +1,196 @@
+	lib:libwx_gtk3u_core_3.1$secondaryArchSuffix = $libVersionCompat
+	lib:libwx_gtk3u_html_3.1$secondaryArchSuffix = $libVersionCompat
+	lib:libwx_gtk3u_propgrid_3.1$secondaryArchSuffix = $libVersionCompat
+	lib:libwx_gtk3u_qa_3.1$secondaryArchSuffix = $libVersionCompat
+	lib:libwx_gtk3u_ribbon_3.1$secondaryArchSuffix = $libVersionCompat
+	lib:libwx_gtk3u_richtext_3.1$secondaryArchSuffix = $libVersionCompat
+	lib:libwx_gtk3u_stc_3.1$secondaryArchSuffix = $libVersionCompat
+	lib:libwx_gtk3u_xrc_3.1$secondaryArchSuffix = $libVersionCompat
+	"
+
+REQUIRES="
+	haiku$secondaryArchSuffix
+	xcairo$secondaryArchSuffix
+	lib:libatk_1.0$secondaryArchSuffix
+	lib:libexecinfo$secondaryArchSuffix
+	lib:libexpat$secondaryArchSuffix
+	lib:libfontconfig$secondaryArchSuffix
+	lib:libfreetype$secondaryArchSuffix
+	lib:libgdk_3$secondaryArchSuffix
+	lib:libgdk_pixbuf_2.0$secondaryArchSuffix
+	lib:libgtk_3$secondaryArchSuffix
+	lib:libglib_2.0$secondaryArchSuffix
+	lib:libharfbuzz$secondaryArchSuffix
+	lib:libiconv$secondaryArchSuffix
+	lib:libintl$secondaryArchSuffix
+	lib:libjpeg$secondaryArchSuffix
+	lib:liblzma$secondaryArchSuffix
+	lib:libpango_1.0$secondaryArchSuffix
+	lib:libpng16$secondaryArchSuffix
+	lib:libSM$secondaryArchSuffix
+	lib:libtiff$secondaryArchSuffix
+	lib:libX11$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	wxgtk${secondaryArchSuffix}_devel = $portVersion
+	cmd:wx_config$secondaryArchSuffix = $portVersion compat >= 3
+	cmd:wxrc$secondaryArchSuffix = $portVersion compat >= 3
+	cmd:wxrc_3.1$secondaryArchSuffix = $libVersionCompat
+	devel:libwx_baseu_3.1$secondaryArchSuffix = $libVersionCompat
+	devel:libwx_baseu_net_3.1$secondaryArchSuffix = $libVersionCompat
+	devel:libwx_baseu_xml_3.1$secondaryArchSuffix = $libVersionCompat
+	devel:libwx_gtk3u_adv_3.1$secondaryArchSuffix = $libVersionCompat
+	devel:libwx_gtk3u_aui_3.1$secondaryArchSuffix = $libVersionCompat
+	devel:libwx_gtk3u_core_3.1$secondaryArchSuffix = $libVersionCompat
+	devel:libwx_gtk3u_html_3.1$secondaryArchSuffix = $libVersionCompat
+	devel:libwx_gtk3u_propgrid_3.1$secondaryArchSuffix = $libVersionCompat
+	devel:libwx_gtk3u_qa_3.1$secondaryArchSuffix = $libVersionCompat
+	devel:libwx_gtk3u_ribbon_3.1$secondaryArchSuffix = $libVersionCompat
+	devel:libwx_gtk3u_richtext_3.1$secondaryArchSuffix = $libVersionCompat
+	devel:libwx_gtk3u_stc_3.1$secondaryArchSuffix = $libVersionCompat
+	devel:libwx_gtk3u_xrc_3.1$secondaryArchSuffix = $libVersionCompat
+	"
+
+REQUIRES_devel="
+	wxgtk$secondaryArchSuffix == $portVersion base
+	xcairo${secondaryArchSuffix}_devel
+	devel:libgdk_3$secondaryArchSuffix
+	devel:libgtk_3$secondaryArchSuffix
+	devel:libjpeg$secondaryArchSuffix
+	devel:libpng16$secondaryArchSuffix
+	devel:libtiff$secondaryArchSuffix
+	devel:libticonv$secondaryArchSuffix
+	devel:libexecinfo$secondaryArchSuffix
+	devel:libexpat$secondaryArchSuffix
+	devel:libiconv$secondaryArchSuffix
+	devel:liblzma$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
```

Comment:
Missing dependencies for the binaries as mentioned in comment: https://github.com/haikuports/haikuports/pull/6637#discussion_r815480656

Reply:
Can you please show me where i can see the build log? because i'm confused about these missing deps. all of these are declared in my local copy and i can see them declared  in this PR too, so i don't know what you mean by missing?

Reply:
@khallebal in strict mode you get policy errors, if not you would see policy warnings like these:

```
creating package wxgtk_x86_devel-3.1.5-1-x86_gcc2.hpkg ...
----- Package Info ----------------
header size:                     80
heap size:                  1578555
TOC size:                     45065
package attributes size:       2026
total size:                 1578635
-----------------------------------
Warning: POLICY ERROR: "/packaging/wxgtk_x86_devel/bin/wxrc-3.1-x86" needs library "libexecinfo.so.1.1", but the package doesn't seem to declare that as a requirement
Warning: POLICY ERROR: "/packaging/wxgtk_x86_devel/bin/wxrc-3.1-x86" needs library "libiconv.so.2", but the package doesn't seem to declare that as a requirement
Warning: POLICY ERROR: "/packaging/wxgtk_x86_devel/bin/wxrc-3.1-x86" needs library "libz.so.1", but the package doesn't seem to declare that as a requirement
Warning: POLICY ERROR: "/packaging/wxgtk_x86_devel/bin/wxrc-3.1-x86" needs library "liblzma.so.5", but the package doesn't seem to declare that as a requirement
Warning: POLICY ERROR: "/packaging/wxgtk_x86_devel/bin/wxrc-3.1-x86" needs library "libexpat.so.1", but the package doesn't seem to declare that as a requirement
Warning: POLICY ERROR: "/packaging/wxgtk_x86_devel/bin/wxrc-x86" needs library "libexecinfo.so.1.1", but the package doesn't seem to declare that as a requirement
Warning: POLICY ERROR: "/packaging/wxgtk_x86_devel/bin/wxrc-x86" needs library "libiconv.so.2", but the package doesn't seem to declare that as a requirement
Warning: POLICY ERROR: "/packaging/wxgtk_x86_devel/bin/wxrc-x86" needs library "libz.so.1", but the package doesn't seem to declare that as a requirement
Warning: POLICY ERROR: "/packaging/wxgtk_x86_devel/bin/wxrc-x86" needs library "liblzma.so.5", but the package doesn't seem to declare that as a requirement
Warning: POLICY ERROR: "/packaging/wxgtk_x86_devel/bin/wxrc-x86" needs library "libexpat.so.1", but the package doesn't seem to declare that as a requirement
Warning: Error: packaging policy violation(s) in strict mode
Error: Build has failed - stopping.
```
As you see it also needs the libs (not just the devel) packages as shown here: https://github.com/haikuports/haikuports/blob/4f180bdb94db9749ae800820b271e262a12754aa/dev-qt/wxqt/wxqt-3.1.3.recipe#L92